pkg/email/config.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package email
+
+import (
+	"context"
+	"fmt"
+
+	"firebase.google.com/go/auth"
+	"github.com/google/exposure-notifications-server/pkg/secrets"
+)
+
+// ProviderType represents a type of email provider.
+type ProviderType string
+
+const (
+	ProviderTypeNoop     ProviderType = "NOOP"
+	ProviderTypeFirebase ProviderType = "FIREBASE"
+	ProviderTypeSMTP     ProviderType = "SIMPLE_SMTP"
+)
+
+// Config represents the env var based configuration for email SMTP server connection.
+type Config struct {
+	ProviderType ProviderType
+
+	User     string `env:"EMAIL_USER" json:",omitempty"`
+	Password string `env:"EMAIL_PASSWORD" json:",omitempty"`
+	SMTPHost string `env:"EMAIL_SMTP_HOST" json:",omitempty"`
```

Comment:
It's probably worth noting that this _won't_ work for most modern Gmail accounts, and won't work for ~G Suite~ Google Workplace either.

Reply:
Wait... but it does. I'm running encvtest@gmail.com to test this

Reply:
You don't have MFA enabled then? It won't work with advanced protection accounts either.

Reply:
https://support.google.com/mail/answer/185833

With MFA you need to create an app password for this to use

Reply:
> App Passwords aren’t recommended and are unnecessary in most cases.

> If you don’t have this option, it might be because:
>
> c. Your account is through work, school, or other organization.
> d. You turned on Advanced Protection.

c is very likely. d will probably become a default in the future.

Reply:
https://github.com/google/exposure-notifications-verification-server/pull/797/commits/72cc99b5a91e929a91aa285b6b1b3aaeb4c338c7

Adding in some notes

Reply:
Thank